src/main/java/network/brightspots/rcv/CsvCvrReader.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * RCTab
+ * Copyright (c) 2017-2022 Bright Spots Developers.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+/*
+ * Purpose: Read and parse generic CSV files
+ * Design: Parses a CSV with candidates in columns, cast vote records in rows,
+ * and vote rankings in cells.
+ * Conditions: The CSV must contain only one contest.
+ * Version history: see https://github.com/BrightSpots/rcv.
+ */
+
+package network.brightspots.rcv;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javafx.util.Pair;
+import network.brightspots.rcv.RawContestConfig.CvrSource;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVRecord;
+
+class CsvCvrReader {
+  private final String cvrPath;
+  private final String contestId;
+  private final String undeclaredWriteInLabel;
+  private final ContestConfig contestConfig;
+  // 0-based column index of first ranking
```

Comment:
If you agree with what @tarheel is saying in the other thread, please update these comments here too.

Reply:
@HEdingfield @tarheel I think the comment is more confusing if we say "1-indexed" -- in my experience, comments above members refer to the members themselves, not their origins or how they were derived.

Thoughts on modifying this so everybody is happy? e.g. `0-based column index of the first ranking, derived by subtracting 1 from the 1-based config variable`? That feels quite long. Or, `1 less than the 1-based column index of the first ranking`? That feels kind of confusing, but is shorter.

Reply:
Given how much confusion this has already generated, I think including the long version is well worth it.